frontend/src/components/Searchbar.astro

```diff
@@ -0,0 +1,264 @@
+---
```

Comment:
It's great that the annoying highlight bug is gone!
The searchbar seems to be less round than before. It also doesn't get hidden when the input field is empty or the user presses 'ESC'

Reply:
Roundness is fixed in https://github.com/oeb25/Cuely/commit/f1905628edadc62785af7736f70e0c7e90eb85bf.

I changed showing/hiding to be determined on focus within the search bar. So to achieve hiding on ESC we need to unfocus the input. I would think it would do this by default, but doesn't seem to be the case. Adding `searchbarInput.blur();` to the `keyup` handler on escape seems to do the job.

Hiding on empty more difficult, but I found a CSS only solution:
```css
hidden group-focus-within:peer-placeholder-shown:hidden group-focus-within:flex
```
where to HTML has been restructured to have `searchInput` as a peer of the suggestions.

The `suggestions-visible` and `searchbar-suggestions-visible` class has become redundant now, apart from some event handlers which have:
```ts
if (!suggestions.classList.contains("suggestions-visible")) {
  return;
}
```
This should be achievable without adding the class. Also, maybe many of these cases can be removed now that visibility is handled purely by CSS?

Reply:
Awesome!
The CSS solution seems great. I must admit it has been a while since I looked thoroughly at that part of the code, so I can't remember which cases we can possibly omit 😅 
I agree that it seems like most of the stuff in the `keyup` handler can be removed, but it would probably be easier to just remove the cases one by one and see if anything breaks

Reply:
I just looked and apparently the searchbar also doesn't auto-hide in the current implementation when the input field is empty. Sorry about that!
Let's just ignore it for now if it is too much of a hassle

Reply:
Check out https://github.com/oeb25/Cuely/commit/d4c159caa68bb0c3642ecce5f093b32d973729b2

Reply:
It's perfect! Awesome!